tests/installation/addon_products_sle.pm

```diff
@@ -88,8 +88,12 @@ sub handle_all_packages_medium {
         next if (skip_package_hub_if_necessary($i));
         push @addons_license_tags, "addon-license-$i" if grep(/^$i$/, @addons_with_license);
         send_key 'home';
-        send_key_until_needlematch "addon-products-all_packages-$i-highlighted", 'down', 30;
-        send_key 'spc';
+        send_key_until_needlematch ["addon-products-all_packages-$i-highlighted", "addon-products-all_packages-$i-selected"], "down", 30;
+        if (match_has_tag("addon-products-all_packages-$i-highlighted")) {
+            send_key 'spc';
+        } else {
```

Comment:
'else' is not very correct here, I think you'd better to use elsif for '"addon-products-all_packages-$i-selected". 'else' means two situation, one is "addon-products-all_packages-$i-selected", one is no tag matched. 
Or just
 if (match_has_tag("addon-products-all_packages-$i-highlighted")) {
         send_key 'spc';
  }
if (match_has_tag("addon-products-all_packages-$i-selected"))
{
}

Reply:
I followed existed code. As send_key_until_needlematch must match one needle. if condition check one needle, else will check other one. So else will not need another condition.